src/exabgp/bgp/message/update/nlri/mvpn/sharedjoin.py

```diff
@@ -0,0 +1,114 @@
+from exabgp.protocol.family import AFI
+from exabgp.protocol.family import SAFI
+
+from exabgp.bgp.message.update.nlri.qualifier import RouteDistinguisher
+from exabgp.bgp.message.update.nlri.mvpn.nlri import MVPN
+from exabgp.bgp.message.notification import Notify
+from exabgp.protocol.ip import IP
+from struct import pack
+
+# +-----------------------------------+
+# |      RD   (8 octets)              |
+# +-----------------------------------+
+# |    Source AS (4 octets)           |
+# +-----------------------------------+
+# | Multicast Source Length (1 octet) |
+# +-----------------------------------+
+# |   Multicast Source (variable)     |
+# +-----------------------------------+
+# |  Multicast Group Length (1 octet) |
+# +-----------------------------------+
+# |  Multicast Group   (variable)     |
+# +-----------------------------------+
+
+
+@MVPN.register
+class SharedJoin(MVPN):
+    CODE = 6
+    NAME = "C-Multicast Shared Tree Join route"
+    SHORT_NAME = "Shared-Join"
+
+    def __init__(self, rd, afi, source, group, source_as, packed=None, action=None, addpath=None):
+        MVPN.__init__(self, afi=afi, action=action, addpath=addpath)
+        self.rd = rd
+        self.group = group
+        self.source = source
+        self.source_as = source_as
+        self._pack(packed)
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, SharedJoin)
+            and self.CODE == other.CODE
+            and self.rd == other.rd
+            and self.source == other.source
+            and self.group == other.group
+        )
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __str__(self):
```

Comment:
Does this represent the route in a way which could be parsed by the configuration code?

Reply:
I didn't get this one. can you share an example?